### PR TITLE
switch to SQL::Abstract::Limit for generating SQL

### PR DESCRIPTION
**Pull request by asb-capfan**
Relates to https://github.com/srchulo/DataTables/issues/1
Also added method table_data to separate JSON serilization from data retrieval, allowing for other interfaces such as tests or maybe UI frameworks. 
Indexcolumns were removed, as not required ( https://github.com/srchulo/DataTables/issues/11 ).
Added query as optional argument to constructor to allow custom query parameters for testing ( https://github.com/srchulo/DataTables/issues/10 ).

### CODE REVIEW

File: lib/DataTables.pm [hunk 1]
```diff
@@ -362,9 +240,128 @@ sub json {
...
 
+sub json {
+    my $self = shift;
+
+    my $output_href = $self->table_data;
+
+    return encode_json $output_href;
+} # /json
+
+
```
Comment (srchulo) [inline lib/DataTables.pm]: Can you remove these extra lines?

Comment (asb-capfan) [reply lib/DataTables.pm]: I'm not sure what you mean. Do you mean the 2 linebreaks after a sub?

**Comment by asb-capfan** [comment]
Hi! Did you have any chance to take a look at the changes proposed?

**Comment by srchulo** [comment]
Sorry for the delay!

**Pull request merged**